### ISSUE TITLE
Incorrect exception message: Insufficient money

### ISSUE THREAD

**Issue by aphex3k** [bisq-network/bisq#6554]
#### Version

v1.9.9

### Description

Bisq app does not compute the funds correctly or fails to take fee to create a transaction even though a sufficient amount of funds is available to cover all positions. This happens several times over.

Im trying to use the funds in my Bisq wallet for this trade.

The app reports an incorrect amount of funds missing.

```
An error occurred at task: CreateTakerFeeTx
Exception message: Insufficient money,  missing 0.00893327 BTC`
```

```
Fund your trade
You need to deposit 0.02875187 BTC (≈ 668.80 USD) + 2.38 BSQ (≈ 2.10 USD) to take this offer.

The amount is the sum of:
- Trade amount: 0.0228 BTC (≈ 530.36 USD) 
- Your security deposit: 0.0057 BTC (≈ 132.59 USD / 25.00% of trade amount)
- Trading fee: 2.38 BSQ (≈ 2.10 USD)
- Total mining fees: 0.00025187 BTC (≈ 5.86 USD / 1.10% of trade amount)

You can choose between two options when funding your trade:
- Use your Bisq wallet (convenient, but transactions may be linkable) OR
- Transfer from an external wallet (potentially more private)

You will see all funding options and details after closing this popup.
```

BTC network and BSQ DAO are in sync. 

Funds available: 0.03043421 BTC
Funds reserved: 0.00 BTC
Funds locked: 0.00 BTC
BSQ available: 236.39

BTC Amount to spend: 0.0228 BTC
Security Deposit: 0.0057 BTC
Mining Fee: 0.00024924 BTC
Total funds required for Trade: 0.02875187

0.03043421 - 0.02875187 = **0.00168234**

Instead I'm getting an error of being short 0.00893327.

### Screenshots

<img width="1470" alt="Screenshot 2023-02-03 at 11 52 14 AM" src="https://user-images.githubusercontent.com/433270/216697589-299cf6be-389c-419a-8876-7dab33ac7002.png">
<img width="277" alt="image" src="https://user-images.githubusercontent.com/433270/216698270-633619d5-0801-43a8-9efa-8e840d8bd606.png">
<img width="787" alt="image" src="https://user-images.githubusercontent.com/433270/216698496-86680304-c3b2-4b09-9dba-c84ec8b22f15.png">
<img width="364" alt="image" src="https://user-images.githubusercontent.com/433270/216698781-5723ed68-2e35-4604-a8c7-c8356fee7cb6.png">
<img width="558" alt="Screenshot 2023-02-03 at 12 04 11 PM" src="https://user-images.githubusercontent.com/433270/216700508-6ac9ae5a-57d9-4e85-960c-e7bf0d7b75d6.png">
<img width="379" alt="Screenshot 2023-02-03 at 12 06 02 PM" src="https://user-images.githubusercontent.com/433270/216700544-55d0d93a-2840-4582-8063-b3d7e36e278c.png">

#### Device or machine

M2 macbook Air, Ventura 13.1 (22C65)

### Additional Notes

Bisq maybe has a problem with unconfirmed transactions as part of the funding source? I can see that at least one transaction in my Bisq wallet is still unconfirmed but because it is non-RBF it counts it as available funds but then they are actually not available when creating the transaction? That would explain the number of 0.00893327 BTC missing which is part of the amount of unconfirmed BTC in my Bisq wallet.

**Comment by ghost** [comment]
Do you have any finished trade for which you have yet to press the "Close" button?  -- might want to close them all.

One other workaround has been to consolidate UTXOs by doing a 2 sat/vB send to self.

**Comment by aphex3k** [comment]
I had no trades that needed to be closed.

**Comment by ghost** [comment]
Had an opportunity to investigate this error recently, notes are as follows for future reference:

- A UTXO gets excluded from the "Available" list due to the address re-use bug (#5725).  i.e. you take an offer and your address containing funds is re-used for the "Trade Payout" address, thus excluding said funds until the trade completes.
- One can verify that by searching for the address in Funds - Receive funds, which equates to the "Available" list.  It should be there, but is not when the address has been re-used for payout by an active trade.
- One can also verify the missing UTXO by `CTRL-J` which lists all addresses' state.
- The same behavior can be observed when sending BSQ - insufficient BTC is available to pay the tx fee.
- The workaround as previously mentioned is doing a send to self which moves funds to a fresh UTXO.  Surprisingly this works even though the address is not marked "Available".


In conclusion, the problem described in #5725 is still causing side-effects.
